compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -168,9 +168,9 @@ impl<'p, 'tcx> Visitor<'p, 'tcx> for MatchVisitor<'p, 'tcx> {
             {
                 let mut chain_refutabilities = Vec::new();
                 let Ok(()) = self.visit_land(ex, &mut chain_refutabilities) else { return };
-                // If at least one of the operands is a `let ... = ...`.
-                if chain_refutabilities.iter().any(|x| x.is_some()) {
-                    self.check_let_chain(chain_refutabilities, ex.span);
+                // Check only single let binding
+                if let [Some((_, refutability))] = chain_refutabilities[..] {
```

Comment:
```suggestion
                if let [Some((_, Irrefutable))] = chain_refutabilities[..] {
```

Could go either way, I suppose, but it almost seems worth just doing the rest of the check inline here.  It's already checking the "only one" part; maybe it should just check the "irrefutable" part also.  Then either `check_single_let` could be renamed to `lint_single_let` (since the check has been moved out of it) or perhaps be fully inlined.

Reply:
I've pushed a new commit addressing this suggestion. Not sure if it meets your expectations

Reply:
Yes, looks right.  With it factored this way, might as well use `lint_single_let` in `check_let` as well.